astropy/coordinates/sky_coordinate.py

```diff
@@ -902,18 +901,18 @@ def to_table(self):
         >>> t.meta
         {'representation_type': 'spherical', 'frame': 'icrs'}
         """
-        self_as_dict = self.info._represent_as_dict()
-        tabledata = {}
-        metadata = {}
-        # Record attributes that have the same length as self as columns in the
-        # table, and the other attributes as table metadata.  This matches
-        # table.serialize._represent_mixin_as_column().
-        for key, value in self_as_dict.items():
+        table = self.frame.to_table()
```

Comment:
It seems a bit strange that do not make use of the fact that we can create the table with exactly the same method as used in `BaseCoordinateFrame`, since the attributes added here would already be dealt with by `_represent_as_dict()`.

Would it be an idea to, instead, follow what was done for `insert()` in #16831 and for now define a private method on the info class that is called by one-line implementations in both `BaseCoordinateFrame` and `SkyCoord`? Alternatively, maybe consider jumping ahead of #17016 and create a mixin class `CoordinateSharedMethods` that defined `to_table()` (and `insert()`? (I'm happy to rebase #17016 on that.)

Reply:
My understanding is that the info classes exist so that coordinates could be used in tables as mixin columns, but the `to_table()` methods are conceptually unrelated to that.

I wouldn't implement a new mixin class just for the `to_table()` methods, but if you are planning to do that for implementing masked attributes anyways then it is indeed worth considering.

Reply:
I agree, and indeed felt somewhat similarly bad about sticking `._insert()` on the info class. At the time, I also felt that just for one method, it was not worth making a mixin class, but now it is two (`insert()` and `to_table()`) and the masked case will add three more.

Reply:
I'm not convinced that we need to put in a lot of effort to unify the implementations of `to_table()` for `SkyCoord` and `BaseCoordinateFrame()`. I would say the docstrings are more onerous to maintain than the code itself, and the docstrings are too different to unify.

Reply:
I'm absolutely fine with postponing any discussion of factoring out common methods to a mixin class until such a mixin class in fact exists. We're definitely risking spending way too much time on a method that has rather little use beyond being a counterpart to `guess_from_table` (indeed, had it been up to me, it would not have existed).